README.md

```diff
@@ -6,106 +8,295 @@
+may be a security risk to enable the import of arbitrary modules when a job is
+enqueued. It's much better to require the module to have been installed (rather
+than just an arbitrary file on a worker's filesystem) in the environment of the
+worker.
 
-```javascript
-// myproject/jobs/generator2job.js
-'use strict';
-const co = require('co');
-
-module.exports = function generator2job(generatorFn) {
-  return {
-    perform(job, cb) {
-      co(generatorFn(job)).then(val => cb(), err => cb(err))
-    }
-  };
-};
+## Running Jobs
+
+There are number of additional bells and whistles when running jobs with
+`qless-js-worker`.
+
+### Worker name (`--name`)
+
+The worker name uniquely identifies the processing entity (the worker). However,
+different deployments may require the use of a workername other than just the
+hostname.
+
+### Concurrency (`--concurrency`)
+
+When running heavily asynchronous workloads, parallelization is possible with
+the `--concurrency` option. It specifies the number of concurrent jobs that can
+be run on a single core.
+
+### Interval (`--interval`)
+
+This configures the polling interval for looking for new jobs. The worker will
+try to pop jobs from the queues it's configured to work from (in round-robin
+order). Only when there are no jobs available does the worker sleep for the
+provided interval before checking again. Otherwise, immediately after completing
+a job, it checks for another.
+
+### Multiple queues (`--queue`)
+
+A worker can grab jobs from multiple queues, as they're available. They're
+popped from the configured queues in round-robin order. To add multiple queues,
+just provide the `--queue` option multiple times:
+
+```bash
+qless-js-worker \
+  --host='redis://my-redis-host:6379' \
+  --queue=queueOne \
+  --queue=queueTwo \
+  --queue=queueThree
 ```
 
-```javascript
-// myproject/jobs/MyJob.js
-'use strict';
+### Verbose logging
 
-const Promise = require('bluebird');
+Use `-v` to enable verbose logging, including information log messages. Use
+`-vv` to enable debugging output as well.
 
-function *doSomeStuff() {
-  console.log("doing some stuff");
-  yield Promise.delay(2000);
-  console.log("I've done some stuff");
-  throw new Error('wombat power!');
-  console.log("I've done some more stuff!");
-}
+## Updating Qless Lua Scripts
+
+This repo contains a submodule to the core `qless` scripts, which can be updated
+with the `build` command:
+
+```bash
+npm run build
+```
+
+## To-Dos
+
+In no particular order of complexity or importance:
+
+- Support a development mode which allows for absolute and relative path imports
+- Sandbox work directory
+- Have an exception hierarchy (Qless exceptions on down)
+- Worker don't fail job on lost lock exception
+- Set the redis instance for testing purposes
+- Add a --verbose flag for more logging. Also -vv
+- Use multiple cores (until then, it's recommended to use parameterized upstart)
+- Support the `--resume` option
+- Recurring job objects
```

Comment:
Sure thing. While we can create recurring jobs, there is no corresponding `RecurringJobs` class to manipulate them. So while I can do this:

```javascript
client.job('my-normal-jid').then((job) => { job.move('another-queue'); })
```

I cannot yet do this:

```javascript
client.job('my-recurring-jid').then((job) => { job.move('another-queue') })
```